extensions/webirc.md

```diff
@@ -0,0 +1,60 @@
+---
+title: IRCv3 WebIRC extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Evan Magaliff"
+    period: "2017"
+    email: "muffinmedic@kiwiirc.com"
+  -
+    name: "Darren Whitlen"
+    period: "2017"
+    email: "prawnsalad@kiwiirc.com"
+---
+## Purpose
+This specification is intended to document the existing `WEBIRC` command that is already in use by multiple IRC servers and WebIRC gateways to provide a standard specification for implementation. This specification does not add any new features or changes, although new features, such as use of public-key cryptography instead of password authentication, can be submitted as extensions.
+
+## Description
+When a user connects through an indirect connection to the IRC server, the user's actual IP address and hostname are not visible. The `WEBIRC` command allows the WebIRC gateway to pass on the user's actual IP address and hostname to the IRC server. This information can be used by the network as if the user was connecting directly.
+
+## Format
+The `WEBIRC` command MUST be the first command sent from the WebIRC gateway to the IRC server and MUST be sent before capability negotiation.
+
+The `WEBIRC` command takes four parameters: `password `, `user`, `hostname`, and `ip`.
+- `password` Password authenticating the WebIRC gateway to the IRC server which must be agreed upon ahead of time
+- `user` WebIRC gateway service name
+- `hostname` Resolved hostname of the user's IP address
```

Comment:
> of the user's IP address

The only thing that can be resolved in this case is an IP address, correct. This clarification is to make it clear that it's the user's IP address being resolved and not the IP of the gateway.